src/pages/Favorites/index.js

```diff
@@ -1,9 +1,55 @@
-const Favorites = () => {
-  return (
-    <>
-      New Favorites
-    </>
-  )
+import React from "react";
+import axios from "axios";
+import { Container, Card, Title, Author, Cover } from "./favorites.styles";
+
+class Favorites extends React.Component {
+  state = {
+    data: [],
+  };
+
+  fetchFavorite = () => {
+    this.props.favorites.map((favorite) => {
+      const retrieveFavorite = async (favorite) => {
+        try {
+          const { data } = await axios(
+            `https://www.googleapis.com/books/v1/volumes/${favorite}`
+          );
+          const updatedState = { ...this.state.data };
+          updatedState[favorite] = data.volumeInfo;
+          this.setState({ data: updatedState });
+        } catch (error) {
+          console.log(error);
+        }
+      };
+      retrieveFavorite(favorite);
+      return favorite;
+    });
+  };
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

Reply:
I'll ask about this in the call